domain/upgrade/state/state.go

```diff
@@ -172,198 +181,163 @@ func (st *State) StartUpgrade(ctx context.Context, upgradeUUID domainupgrade.UUI
-// all provisioned controller, the upgrade itself will be completed.
+// SetControllerDone marks the supplied controllerID as having completed its
+// upgrades. When SetControllerDone is called by the all provisioned controller,
+// the upgrade itself will be completed.
 //
-// TODO (jack-w-shaw) Set `statuses`/`statuseshistory` here
-// to status.Available when we complete an upgrade
+// TODO (jack-w-shaw) Set `statuses`/`statuseshistory` here to status.Available
+// when we complete an upgrade
 func (st *State) SetControllerDone(ctx context.Context, upgradeUUID domainupgrade.UUID, controllerID string) error {
 	db, err := st.DB()
 	if err != nil {
 		return errors.Trace(err)
 	}
 
-	lookForDoneNodesQuery := `
-SELECT (controller_node_id, node_upgrade_completed_at) AS (&infoControllerNode.*)
+	controllerNodeInfo := ControllerNodeInfo{
+		UpgradeInfoUUID:  upgradeUUID.String(),
+		ControllerNodeID: controllerID,
+	}
+	info := Info{
+		UUID: upgradeUUID.String(),
+	}
+
+	lookForDoneNodesStmt, err := st.Prepare(`
+SELECT (controller_node_id, node_upgrade_completed_at) AS (&ControllerNodeInfo.*)
 FROM   upgrade_info_controller_node
-WHERE  upgrade_info_uuid = $M.info_uuid
-AND    controller_node_id = $M.controller_id;`
-	lookForDoneNodesStmt, err := st.Prepare(lookForDoneNodesQuery, infoControllerNode{}, sqlair.M{})
+WHERE  upgrade_info_uuid = $ControllerNodeInfo.upgrade_info_uuid
+AND    controller_node_id = $ControllerNodeInfo.controller_node_id;
+`, controllerNodeInfo)
 	if err != nil {
-		return errors.Annotatef(err, "preparing select done query")
+		return errors.Annotate(err, "preparing select done query")
 	}
 
-	setNodeToDoneQuery := `
+	setNodeToDoneStmt, err := st.Prepare(`
 UPDATE  upgrade_info_controller_node
 SET     node_upgrade_completed_at = DATETIME("now")
-WHERE   upgrade_info_uuid = $M.info_uuid
-AND     controller_node_id = $M.controller_id
+WHERE   upgrade_info_uuid = $ControllerNodeInfo.upgrade_info_uuid
+AND     controller_node_id = $ControllerNodeInfo.controller_node_id
 AND     node_upgrade_completed_at IS NULL;
-`
-	setNodeToDoneStmt, err := st.Prepare(setNodeToDoneQuery, sqlair.M{})
+`, controllerNodeInfo)
 	if err != nil {
 		return errors.Annotatef(err, "preparing update node query")
 	}
 
-	completeUpgradeQuery := `
+	m := sqlair.M{
+		"from_state": upgrade.DBCompleted,
+		"to_state":   upgrade.StepsCompleted,
+	}
+	completeUpgradeStmt, err := st.Prepare(`
 UPDATE upgrade_info
 SET    state_type_id = $M.to_state
-WHERE  uuid = $M.info_uuid AND state_type_id = $M.from_state
+WHERE  uuid = $Info.uuid AND state_type_id = $M.from_state
 AND (
     SELECT COUNT(*)
 	FROM   upgrade_info_controller_node
-    WHERE  upgrade_info_uuid = $M.info_uuid
+    WHERE  upgrade_info_uuid = $Info.uuid
     AND    node_upgrade_completed_at IS NOT NULL
 ) = (
     SELECT COUNT(*) 
 	FROM   upgrade_info_controller_node
-    WHERE  upgrade_info_uuid = $M.info_uuid
+    WHERE  upgrade_info_uuid = $Info.uuid
```

Comment:
Mixing tabs and spaces in multi-line SQL string

Reply:
Its a lot more obvious on github haha.